android/src/main/java/com/swmansion/rnscreens/ScreenStackFragment.kt

```diff
@@ -99,50 +140,357 @@ class ScreenStackFragment :
+                                    screen.sheetDetents.count(),
+                                )
+                            skipCollapsed = false
+                            isFitToContents = false
+                            peekHeight = (screen.sheetDetents[0] * containerHeight).toInt()
+                            expandedOffset = ((1 - screen.sheetDetents[2]) * containerHeight).toInt()
+                            halfExpandedRatio =
+                                (screen.sheetDetents[1] / screen.sheetDetents[2]).toFloat()
+                        }
+
+                    else -> throw IllegalStateException(
+                        "[RNScreens] Invalid detent count ${screen.sheetDetents.count()}. Expected at most 3.",
                     )
+                }
             }
 
-        view?.addView(appBarLayout)
-        if (isToolbarShadowHidden) {
-            appBarLayout?.elevation = 0f
-            appBarLayout?.stateListAnimator = null
+            is KeyboardVisible -> {
+//                val newMaxHeight = max(1, behavior.maxHeight - keyboardState.height)
+                val newMaxHeight =
+                    if (behavior.maxHeight - keyboardState.height >
+                        1
+                    ) {
+                        behavior.maxHeight - keyboardState.height
+                    } else {
+                        behavior.maxHeight
+                    }
+                when (screen.sheetDetents.count()) {
+                    1 ->
+                        behavior.apply {
+                            state = BottomSheetBehavior.STATE_EXPANDED
+                            skipCollapsed = true
+                            isFitToContents = true
+                            maxHeight = newMaxHeight
+                            addBottomSheetCallback(keyboardSheetCallback)
+                        }
+
+                    2 ->
+                        behavior.apply {
+                            state = BottomSheetBehavior.STATE_EXPANDED
+                            skipCollapsed = false
+                            isFitToContents = true
+                            maxHeight = newMaxHeight
+                            addBottomSheetCallback(keyboardSheetCallback)
+                        }
+
+                    3 ->
+                        behavior.apply {
+                            state = BottomSheetBehavior.STATE_EXPANDED
+                            skipCollapsed = false
+                            isFitToContents = false
+                            maxHeight = newMaxHeight
+                            addBottomSheetCallback(keyboardSheetCallback)
+                        }
+
+                    else -> throw IllegalStateException(
+                        "[RNScreens] Invalid detent count ${screen.sheetDetents.count()}. Expected at most 3.",
+                    )
+                }
+            }
+
+            is KeyboardDidHide -> {
```

Comment:
Not really. This is the same `is` operator you usually use to check whether an object is an instance of given class, e.g. `if (view is Screen) ...`. It can be used also in `when` structure in pattern matching context.